cryostatio/cryostat-web
# chore(aa): automated analysis card refactoring

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #751
Fixes: #839

## Description of the change:
Changes include:
* [x] Refactoring `AutomatedAnalysisConfigForm` such that it no longer needs buttons to save the config to local storage
* [x] Adding translation keys to various needed areas of `AutomatedAnalysisCard` and related components
* [x] Adding an inline view of Automated Analysis results
* [x] Fixed rendering bug with the AA Card where filters and toolbar render and flicker before analysis is fetched

Misc:
* [x] Temporary default localized keys now default with the value `cryostat_tmp`. This is so it's easier to search for the string and find the default keys in the file to fix them.  
* [x] Change missing import eslint rule:  `error` -> `warn`
* [x] Exclude `dist` from jest testing

In draft mode. Full changes and tests have not been fixed yet.

## How to manually test:
1. Run cryostat normally with the github action.
2. Add the AutomatedAnalysisCard with the Dashboard AddCard feature and try the configuration editing as well.
3. Create an automated analysis recording and switch the switch to list view to see the new view.
